cnf-certification-test/observability/suite.go

```diff
@@ -17,14 +17,88 @@
 package observability
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
 	"github.com/onsi/ginkgo/v2"
 	"github.com/sirupsen/logrus"
 	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/core/v1"
 )
 
 //
 // All actual test code belongs below here.  Utilities belong above.
 //
 var _ = ginkgo.Describe(common.ObservabilityTestKey, func() {
-	logrus.Debugf("%s not moved yet to new framework", common.ObservabilityTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		provider.BuildTestEnvironment()
+		env = provider.GetTestEnvironment()
+	})
+
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestLoggingIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testContainersLogging(&env)
+	})
 })
+
+// containerHasLoggingOutput helper function to get the last line of logging output from
+// a container. Returns true in case some output was found, false otherwise.
+func containerHasLoggingOutput(cut *provider.Container) (bool, error) {
+	ocpClient := clientsholder.NewClientsHolder()
+
+	numLogLines := int64(1)
+	podLogOptions := v1.PodLogOptions{TailLines: &numLogLines, Container: cut.Data.Name}
+	req := ocpClient.Coreclient.Pods(cut.Namespace).GetLogs(cut.Podname, &podLogOptions)
+
+	podLogsReaderCloser, err := req.Stream(context.TODO())
+	if err != nil {
+		return false, fmt.Errorf("unable to get log streamer, err: %s", err)
+	}
+
+	defer podLogsReaderCloser.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogsReaderCloser)
+	if err != nil {
+		return false, fmt.Errorf("unable to get log data, err: %s", err)
+	}
+
+	return buf.String() != "", nil
+}
+
+func testContainersLogging(env *provider.TestEnvironment) {
+	if len(env.Containers) == 0 {
+		ginkgo.Skip("No containers to run test, skipping")
+	}
+
+	// Iterate through all the CUTs to get their log output. The TC checks that at least
+	// one log line is found.
+	badContainers := []string{}
+	for _, cut := range env.Containers {
+		ginkgo.By(fmt.Sprintf("Checking container %s (pod %s, ns %s) has some logging output", cut.Data.Name, cut.Podname, cut.Namespace))
```

Comment:
optionally, use the StringShort function to display cut (from merge)

Reply:
Done.